tests/agent_server/test_server_agent.py

```diff
@@ -485,6 +501,53 @@ async def verify_deployment_result():
     await resource_action_consistency_check()
 
 
+async def test_env_setting_wiring_to_autostarted_agent(
```

Comment:
Why does this test case set the `AUTOSTARTED_AGENT_MAP` and why does it start an agent? This is not required to test the behavior of `_make_agent_config()`, is it?